### PR TITLE
Upgrade mobx, mobx-react and mobx-state-tree

### PR DESCRIPTION
**Pull request by eatyourgreens**
Update the NextJS apps to use mobx@5.15, mobx-react@6.3 and mobx-state-tree@3.17. 
Upgrade the classifier to use mobx@5.15, mobx-react@6.1 and mobx-state-tree@3.17. 
Upgrading mobx-react to 6.3 breaks the classifier tests, so I've left it alone for now.
Rewrite the project preferences store tests to [use `mobx.when`](https://github.com/mobxjs/mobx/issues/494#issuecomment-241256525). Previously, the tests were run without waiting for the project preferences to load, which broke in mobx-state-tree 3.16 and higher.
Also fixes a couple of project preferences tests which were accidentally nested inside another test (misplaced `})` somewhere) and a typo in the same tests (`projects.createActive(project.id)` instead of `projects.setActive(project.id)`.)

See #1839 for the major bug that appeared the last time I tried to upgrade the NextJS apps to `mobx-react` v6. I haven't been able to reproduce it on this branch, testing locally with a production build and PH-TESS.

Package:
app-content-pages
app-project
lib-classifier

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
The last attempt to upgrade mobx-react broke classifications for live projects (#1839.) I'm hoping this will avoid that by installing the same versions of mobx and mobx-react across the NextJS apps and the classifier.

**Comment by eatyourgreens** [comment]
I'm not seeing a recurrence of #1839 when I submit empty classifications on this branch.
http://local.zooniverse.org:3000/projects/nora-dot-eisner/planet-hunters-tess/classify/workflow/11235?demo=true

**Comment by eatyourgreens** [comment]
Should we upgrade `mobx` from 5 to 6? [Migration guide](https://mobx.js.org/migrating-from-4-or-5.html).

**Comment by eatyourgreens** [comment]
UPP tests are all broken because they don't wait for the UPP to load before running the expectations. 😞

File: packages/lib-classifier/src/store/UserProjectPreferencesStore.spec.js [hunk 1]
```diff
@@ -76,11 +89,18 @@ describe('Model > UserProjectPreferencesStore', function () {
     it('should set project preferences if there is a user and a project', function (done) {
       rootStore = setupStores(clientStub, authClientStubWithUser)
       rootStore.projects.setActive(project.id)
-        .then(() => {
+      function preferencesAreLoaded() {
+        return rootStore.userProjectPreferences.loadingState === asyncStates.success
+      }
+      when(
+        preferencesAreLoaded,
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/UserProjectPreferencesStore.spec.js:96]: This delays running the tests until `preferencesAreLoaded()` is true.

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Projects: `app-project` `lib-classifier` `app-content-pages`

This PR is a major dependency update for our app's stores.

- In general, testing looks good. I'm unable to break any of the main functionality of the app(s)
- I'm unable to replicate the error in #1839 - submitting a Classification with no annotations works fine. This is good 👍 

### Testing

**Part 1:** Tested `app-project` on macOS10+Chrome88 on `localhost:3000`

Looking good:

- With a workflow with 1 Single Question Task, I'm able to...
  - ...answer one question and submit.
  - ...provide no answer and submit
- With a workflow with 2 Single Question Tasks, I'm able to...
  - ...answer two questions and submit.
  - ...answer one of the questions and submit.
  - ...provide no answer and submit
  - ...go back to a previous question to amend an answer
- With a workflow with 1 Drawing Task (Point tool, Point tool), I'm able to...
  - ...add, edit, and delete Point marks. 
  - ...provide no marks and submit.
  - ...provide 1-5 marks and submit.
- With a workflow with 1 Drawing Task (Point tool, Point tool), and 1 Text Task, I'm able to...
  - ...add, edit, and delete Point marks.
  - ...proceed to the Text Task, then go back to the Drawing Task to amend my earlier annotations.
    - ⚠️ [minor quirk] there's an oddity where going back to a previous drawing task causes the previous annotations to STILL appear as untouchable "previous marks", but this is extant even in `master` and not related to this PR. Probably a different bug to fix. 
  - ...provide any combination of answer/non-answers and submit

Additional testing on Safari12 and Win+Edge88 _(but Edge is basically Chrome in a fancy hat at this moment so no surprise that there are no surprises here.)_

**Part 2:** Tested `lib-classifier` on `local.zooniverse.org:8080`

- Basic tests for lib-classifier looks fine.
- Based on passing tests in Part 1, I did not feel the need to test the lib-classifier alone too thoroughly.

**Part 3:** Tested `app-content-pages` on macOS10+Chrome88 on `localhost:3000`

- Content pages loaded fine and functioned normally.

### Testing Limitations

- I'm unable to test authentication/login on  `app-project` on Chrome due to issues accessing `https://local.zooniverse.org:3000` on my end. (Chrome spews SSL errors for my pseudo-localhost)
  - On _Safari_ (which is far more permissive with SSL infractions), however, I can confirm that auth/login works fine, for what that's worth.

### Status

This PR looks good to me. 👍

**Comment by shaunanoordin** [comment]
> ⚠️ [minor quirk] there's an oddity where going back to a previous drawing task causes the previous annotations to STILL appear as untouchable "previous marks", but this is extant even in master and not related to this PR. Probably a different bug to fix.

I just realised you opened #2011 precisely for this. Reviewing that now!

**Pull request merged**